google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -87,24 +110,81 @@ public boolean isSupportAlreadyAdded(@NotNull Module module,
     VirtualFile[] contentRoots = rootModel.getContentRoots();
     AppEngineProjectService appEngineProjectService = AppEngineProjectService.getInstance();
     if (contentRoots.length > 0) {
-      facet.getConfiguration().setAppYamlPath(
+      Path appYamlPath = Paths.get(
           appEngineProjectService.getDefaultAppYamlPath(contentRoots[0].getPath()));
-      facet.getConfiguration().setDockerfilePath(
+      Path dockerfilePath = Paths.get(
           appEngineProjectService.getDefaultDockerfilePath(contentRoots[0].getPath()));
+
+      facet.getConfiguration().setAppYamlPath(appYamlPath.toString());
+      facet.getConfiguration().setDockerfilePath(dockerfilePath.toString());
+
+      // Configuration file generation.
+      Project project = facet.getModule().getProject();
+      Optional<Path> defaultAppYaml =
+          new CloudSdkAppEngineHelper(project)
+              .defaultAppYaml(FlexibleRuntime.java);
+
+      if (generateConfigFiles) {
+        if (Files.exists(appYamlPath)) {
+          int override = Messages.showYesNoDialog(project,
+              GctBundle.message("appengine.support.appyaml.existing"),
+              GctBundle.message("appengine.support.appyaml.existing.title"),
+              GoogleCloudToolsIcons.APP_ENGINE
+          );
+
+          if (override == Messages.YES) {
+            defaultAppYaml.ifPresent(
+                appYaml -> WriteCommandAction.runWriteCommandAction(project,
+                    () -> {
+                      VirtualFile appYamlVirtualFile = contentRoots[0]
+                          .findFileByRelativePath("/src/main/appengine/app.yaml");
+                      if (appYamlVirtualFile != null) {
+                        PsiFile appYamlPsiFile =
+                            PsiManager.getInstance(project).findFile(appYamlVirtualFile);
+                        if (appYamlPsiFile != null) {
+                          Document appYamlDocument =
+                              PsiDocumentManager.getInstance(project)
+                                  .getDocument(appYamlPsiFile);
+                          if (appYamlDocument != null) {
+                            try {
+                              appYamlDocument.setText(
+                                  new String(Files.readAllBytes(appYaml),
+                                      Charset.defaultCharset()));
+                            } catch (IOException ioe) {
+                              logger.debug("Could not copy app.yaml text. " + ioe.getMessage());
+                            }
+                          }
+                        }
+                      }
+                    }));
+          }
+        } else { // !Files.exists(appYamlPath)
+          // Just copy the file.
+          defaultAppYaml.ifPresent(
+              appYaml -> {
+                try {
+                  FileUtil.copy(appYaml.toFile(), appYamlPath.toFile());
```

Comment:
Not when the file didn't exist before, but I'm glad you mention it because it's in my immediate plans to look into using Documents/PsiFiles to generate this file too.